### PR TITLE
bluetooth: mesh: Fix IVU duration counter update

### PR DESCRIPTION
**Pull request by omkar3141**
When device is first provisioned with IV Update
flag is set to 0, it should wait for minimum of
96 hours before going into IV Update In Progress
state. Such limit does not apply, if device is
provisioned with IV Update flag is set to 1.

Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>

### CODE REVIEW

File: tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c [hunk 1]
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 #include "mesh_test.h"
+#include <mesh/net.h>
```
Comment (trond-snekvik) [inline tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_provision.c:7]: Should be `"mesh/net.h"`, as only the include folder is treated as a system header directory

File: subsys/bluetooth/mesh/net.c [hunk 2]
```diff
@@ -184,11 +184,13 @@ int bt_mesh_net_create(uint16_t idx, uint8_t flags, const uint8_t key[16],
 	atomic_set_bit_to(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS,
 			  BT_MESH_IV_UPDATE(flags));
 
-	/* Set minimum required hours, since the 96-hour minimum requirement
-	 * doesn't apply straight after provisioning (since we can't know how
-	 * long has actually passed since the network changed its state).
+	/* If IV Update is already in progress, set minimum required hours,
+	 * since the 96-hour minimum requirement doesn't apply in this case straight
+	 * after provisioning.
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/net.c:189]: Good catch :+1: Would you mind adding a reference to the specification section for this here (or in the commit message)? Should be 3.11.5

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
Nice change!

Would you mind splitting the fix and the new test cases into two different commits? The change to the existing test could stay in the fix commit

**Review by trond-snekvik — APPROVED** [review]

File: subsys/bluetooth/mesh/net.c (hunk 2 above)
Comment (LingaoM) [reply subsys/bluetooth/mesh/net.c:189]: I'm confused. The definition in `Mesh spec 1.0.1 section 5 provision 5.4.2.5 distribution of provision data` seems to have another meaning?

`Note: After the processing the Provisioning Data PDU from the Provisioner, the 96 hour time limits for changing the IV Update procedure state, as defined in the IV Update procedure, do not apply.`

**Review by LingaoM — APPROVED** [review]

File: subsys/bluetooth/mesh/net.c (hunk 2 above)
Comment (omkar3141) [reply subsys/bluetooth/mesh/net.c:189]: Yes, this is slightly wrong and underspecified in the published. This should get rectified in the spec eventually.

**Review by alxelax — APPROVED** [review]

**Pull request merged**